eox_tenant/signals.py

```diff
@@ -217,3 +225,97 @@ def clear_tenant(sender, request, **kwargs):  # pylint: disable=unused-argument
     Signal: django.core.signals.got_request_exception
     """
     pass
+
+
+def before_task_publish(sender, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Dispatched before a celery task is published. Note that this is executed in the process sending the task.
+    See:
+        https://celery.readthedocs.io/en/latest/userguide/signals.html#before-task-publish
+    """
+    headers = kwargs.get('headers')
+    body = kwargs.get('body')
+    try:
+        request_info = body.get('args')[1].get('request_info')
+        headers['eox_tenant_sender'] = request_info.get('host') or None
+    except AttributeError:
+        # This is a async taks generated outside of the scope of a request i.e. CLI task
+        get_host_func = AsyncTaskHandler().get_host_from_task(sender)
+        headers['eox_tenant_sender'] = get_host_func(body)
+
+
+def task_prerun(sender, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Dispatched before a task is executed.
+    See:
+       https://celery.readthedocs.io/en/latest/userguide/signals.html#task-prerun
+    """
+    headers = sender.request.get('headers', {})
+
+    http_host = headers.get('eox_tenant_sender', None)
+
+    if not http_host:
+        LOG.warning("Could not find the host information for eox_tenant.signals")
+
+    domain = http_host.split(':')[0]
```

Comment:
I think that in the previous if, I should set that variable as the default http_host, what do you think?

Reply:
maybe something like this `http_host = headers.get('eox_tenant_sender',  default_host(body))` in line 255, you could delete the if statement  because you log that in line 307, but I don't know  if you can get the body here

Reply:
I think it's better to  set the default value inside the if statement @morenol  forget the previous message